### PR TITLE
ImageSecurityPolicy generated code and scripts

### LINKED ISSUE #20: Setup codegen scripts for CRD clients

### PR DESCRIPTION
**Pull request by priyawadhwa**
Should fix #20

### CODE REVIEW

File: pkg/kritis/apis/kritis/v1beta1/types.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ImageSecurityPolicy is a specification for a ImageSecurityPolicy resource
+type ImageSecurityPolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec                 ImageSecurityPolicySpec `json:"spec"`
+	RequiredAttestations []string                `json:"requiredAttestations"`
```
Comment (tejal29) [inline pkg/kritis/apis/kritis/v1beta1/types.go]: What is this field for?

File: pkg/kritis/apis/kritis/register.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kritis
+
+const (
+	GroupName = "kritis.github.com"
```
Comment (dlorenc) [inline pkg/kritis/apis/kritis/register.go]: maybe "kritis.grafeas.io"

File: pkg/kritis/apis/kritis/v1beta1/zz_generated.deepcopy.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
+// +build !ignore_autogenerated
```
Comment (dlorenc) [inline pkg/kritis/apis/kritis/v1beta1/zz_generated.deepcopy.go]: If you add a .gitattributes file to the top level, you can make these be ignored in github diffs:

```

# This file is documented at https://git-scm.com/docs/gitattributes.
# Linguist-specific attributes are documented at
# https://github.com/github/linguist.

**/zz_generated.*.go linguist-generated=true
/pkg/client/** linguist-generated=true
```

File: hack/update-codegen.sh [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+SCRIPT_ROOT=$(dirname ${BASH_SOURCE})/..
+CODEGEN_PKG=${CODEGEN_PKG:-$(cd ${SCRIPT_ROOT}; ls -d -1 ./vendor/k8s.io/code-generator 2>/dev/null || echo ${GOPATH}/src/k8s.io/code-generator)}
+
+vendor/k8s.io/code-generator/generate-groups.sh all \
```
[the remark this answers is not in the document]
Comment (priyawadhwa) [reply hack/update-codegen.sh]: yup updated to generate deepcopy functions, clientsets, and a lister for imagesecuritypolicies

File: pkg/kritis/apis/kritis/v1beta1/zz_generated.deepcopy.go (hunk 3 above)
Comment (priyawadhwa) [reply pkg/kritis/apis/kritis/v1beta1/zz_generated.deepcopy.go]: awesome, done!

**Pull request merged**